src/exec/util/file.go

```diff
@@ -15,20 +15,119 @@
 package util
 
 import (
+	"bytes"
+	"compress/gzip"
+	"errors"
 	"io/ioutil"
+	"net/http"
 	"os"
 	"path/filepath"
 
 	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/src/log"
+	"github.com/coreos/ignition/src/util"
+
+	"github.com/coreos/ignition/third_party/github.com/vincent-petithory/dataurl"
 )
 
 const (
-	DefaultDirectoryPermissions types.FileMode = 0755
-	DefaultFilePermissions      types.FileMode = 0644
+	DefaultDirectoryPermissions os.FileMode = 0755
+	DefaultFilePermissions      os.FileMode = 0644
+)
+
+var (
+	ErrSchemeUnsupported = errors.New("unsupported source scheme")
+	ErrStatusBad         = errors.New("bad HTTP response status")
 )
 
+type File struct {
+	Path     types.Path
+	Contents []byte
+	Mode     os.FileMode
+	Uid      int
+	Gid      int
+}
+
+func RenderFile(l *log.Logger, f types.File) *File {
```

Comment:
Yeah... take two. This is a lot nicer.

The logger provides a lot of nice functionality like nested contexts. The plan is to move to capnslog eventually, but it's a pretty low priority.